### ISSUE TITLE
Empty Response Causes Failure

### ISSUE THREAD

**Issue by jchblsc** [watson-developer-cloud/assistant-simple#184]
This code in app.js fails when Watson Assistant has no response.  From a node with no response text.  Node v7.10.1.

```javascript
if (data.output.text.length === 0) {
      if (data.output.generic !== undefined) {
        if (data.output.generic[0].text !== undefined) {
          data.output.text = data.output.generic[0].text;
        } else if (data.output.generic[0].title !== undefined) {
          data.output.text = data.output.generic[0].title;
        }
      }
}
```

with error: 

```
C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\src\app.js:62
        if (data.output.generic[0].text !== undefined) {
                                  ^

TypeError: Cannot read property 'text' of undefined
    at C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\src\app.js:62:35
    at Request._callback (C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\node_modules\watson-developer-cloud\lib\requestwrapper.js:125:9)
    at Request.self.callback (C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\node_modules\request\request.js:185:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:194:7)
    at Request.<anonymous> (C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\node_modules\request\request.js:1157:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:191:7)
    at Gunzip.<anonymous> (C:\jch\Prog\JavaScript\eclipse\workspace\JchWatsonAsstAppleQuery\node_modules\request\request.js:1079:12)
    at Object.onceWrapper (events.js:293:19)
```

Also the assumption that there will be only one text response from Watson Assistant is problematic as there may be multiple text responses.  This code will miss anything other than the first response.  This could happen when a node has 'skip user input' with no response text and then the subsequent node has response text.  Like this:

```json
"output": {
    "generic": [
      {
        "response_type": "text",
        "text": ""
      },
      {
        "response_type": "text",
        "text": "You have 10,234 assets in all sites."
      }
    ],
    "text": [
      "",
      "You have 10,234 assets in all sites."
    ],
```
I was able to side-step the immediate problem with this loadash-based code: 

```javascript
if (data.output.text.length === 0 && _.has(data.output, 'generic')) 
    	if (_.isArray(data.output.generic)) 
	        if (_.has(data.output.generic[0], 'text')) 
	        	data.output.text = data.output.generic[0].text;
	        else 
	        	if (_.has(data.output.generic[0]), 'title') 
	        		data.output.text = data.output.generic[0].title;
```
But does not solve the multi-response situation.  

I guess after all this, the corollary question is - does 'generic' text ever show up without a corresponding `data.output.text`?

**Comment by germanattanasio** [comment]
@NEigenfeld Can you look at this?

**Comment by noah-eigenfeld** [comment]
In the case that the application returns an empty response, should there be some output anyway to indicate to the user that they should not wait for a response? 

For example, in the function updateMessage in app.js, we update the response text based on the confidence of the response. If confidence is too low, the response gets changed to "I did not understand your intent." Should a similar message be added if the response text is blank?

Here's an example of what the conversation looks like when a blank response is returned (after the user inputs "jazz"):
<img width="417" alt="screen shot 2018-08-21 at 10 26 53 am" src="https://user-images.githubusercontent.com/11284492/44407793-f6a5b880-a52c-11e8-8692-2e6e7ffb721a.png">
and here's the corresponding data.output object:
`Data.output:
{ generic: [],
  text: [],
  nodes_visited: [ 'node_4_1530146887532' ],
  log_messages: [] }
------------
Output text:
[]`

Maybe this is outside the scope of this issue... @germanattanasio what are your thoughts?

**Comment by germanattanasio** [comment]
fixed in #189
